### PR TITLE
Allow customization of snapshot agent interval

### PR DESCRIPTION
**Pull request by rescDev**
Changes proposed in this PR:
- enable possibility to set custom interval for snapshot agent
- default set to `1h` as stated in the docs https://www.consul.io/commands/snapshot/agent#interval

How I've tested this PR:
unit tests

How I expect reviewers to test this PR:
the same way as I did ;)

Checklist:
- [x] Tests added
- [ ] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: charts/consul/values.yaml [hunk 1]
```diff
@@ -51,7 +51,6 @@ global:
       # If not set and using a NodePort service, Kubernetes will automatically assign
       # a port.
       nodePort:
-
```
Comment (lkysow) [inline charts/consul/values.yaml:54]: please don't reformat this file

File: charts/consul/values.yaml [hunk 2]
```diff
@@ -1296,6 +1292,11 @@ client:
     # The number of snapshot agents to run.
     replicas: 2
 
+    # Interval to perform a snapshot run, defaults to 1h
+    # More infos on the format https://www.consul.io/commands/snapshot/agent#interval
+    # @type: string
+    interval: 1h
```
Comment (lkysow) [inline charts/consul/values.yaml:1298]:
```suggestion
    # Interval at which to perform snapshots.
    # See https://www.consul.io/commands/snapshot/agent#interval
    # @type: string
    interval: 1h
```

File: charts/consul/test/unit/client-snapshot-agent-deployment.bats [hunk 3]
```diff
@@ -1071,3 +1071,24 @@ MIICFjCCAZsCCQCdwLtdjbzlYzAKBggqhkjOPQQDAjB0MQswCQYDVQQGEwJDQTEL' \
   actual=$(echo $object | jq -r '.metadata.annotations["vault.hashicorp.com/role"]' | tee /dev/stderr)
   [ "${actual}" = "sa-role" ]
 }
+
+@test "client/SnapshotAgentDeployment: interval property default" {
```
Comment (lkysow) [inline charts/consul/test/unit/client-snapshot-agent-deployment.bats:1075]:
```suggestion
@test "client/SnapshotAgentDeployment: interval defaults to 1h" {
```

File: charts/consul/test/unit/client-snapshot-agent-deployment.bats [hunk 4]
```diff
@@ -1071,3 +1071,24 @@ MIICFjCCAZsCCQCdwLtdjbzlYzAKBggqhkjOPQQDAjB0MQswCQYDVQQGEwJDQTEL' \
...
+  cd `chart_dir`
+  local actual=$(helm template \
+      -s templates/client-snapshot-agent-deployment.yaml  \
+      --set 'client.snapshotAgent.enabled=true' \
+      . | tee /dev/stderr |
+      yq '[.spec.template.spec.containers[0].command[2]] | any(contains("-interval=1h"))' | tee /dev/stderr)
```
Comment (lkysow) [inline charts/consul/test/unit/client-snapshot-agent-deployment.bats:1081]: See similar tests above:

```
yq -r '.spec.template.spec.containers[0].command[2] | contains("-config-dir=/consul/config")' | tee /dev/stderr)
```

You don't need the `yq '[`

File: charts/consul/test/unit/client-snapshot-agent-deployment.bats [hunk 5]
```diff
@@ -1071,3 +1071,24 @@ MIICFjCCAZsCCQCdwLtdjbzlYzAKBggqhkjOPQQDAjB0MQswCQYDVQQGEwJDQTEL' \
...
+  [ "${actual}" = "true" ]
+}
+
+@test "client/SnapshotAgentDeployment: interval property is overwritten" {
```
Comment (lkysow) [inline charts/consul/test/unit/client-snapshot-agent-deployment.bats:1085]:
```suggestion
@test "client/SnapshotAgentDeployment: interval can be set" {
```

**Comment by rescDev** [comment]
thanks for the review @lkysow :) I think of all of your comments have been addressed

**Review by lkysow — APPROVED** [review]
Thanks so much!

**Review by kschoche — APPROVED** [review]
Thanks for this PR, it looks great!

**Pull request merged**